#1091 causes build failure

4aae411d2f432caae887fd2b6a595f18e8c71022 from #1091 reliably causes the local Flatpak build to fail when building EasyEffects. Using the previous commit f5ef2adea626dcde28f306c626067e5d7cf2b5cf is fine.
```
[81/105] Compiling C++ object src/easyeffects.p/plugin_ui_base.cpp.o
FAILED: src/easyeffects.p/plugin_ui_base.cpp.o 
ccache c++ -Isrc/easyeffects.p -Isrc -I../src -I../include -I. -I.. -Idata -I/app/include/pipewire-0.3 -I/app/include/spa-0.2 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/app/include/glibmm-2.68 -I/app/lib/glibmm-2.68/include -I/app/include/sigc++-3.0 -I/app/lib/sigc++-3.0/include -I/usr/include/gtk-4.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/graphene-1.0 -I/usr/lib/x86_64-linux-gnu/graphene-1.0/include -I/usr/include/gio-unix-2.0 -I/app/include/gtkmm-4.0 -I/app/lib/gtkmm-4.0/include -I/app/include/giomm-2.68 -I/app/lib/giomm-2.68/include -I/app/include/cairomm-1.16 -I/app/lib/cairomm-1.16/include -I/app/include/pangomm-2.48 -I/app/lib/pangomm-2.48/include -I/usr/include/gtk-4.0/unix-print -I/app/include/lilv-0 -I/app/include/serd-0 -I/app/include/sord-0 -I/app/include/sratom-0 -I/app/include/bs2b -I/app/include/ -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++20 -O2 -g '-DG_LOG_DOMAIN="easyeffects"' -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -pthread -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 -D_REENTRANT -MD -MQ src/easyeffects.p/plugin_ui_base.cpp.o -MF src/easyeffects.p/plugin_ui_base.cpp.o.d -o src/easyeffects.p/plugin_ui_base.cpp.o -c ../src/plugin_ui_base.cpp
../src/plugin_ui_base.cpp: In static member function ‘static float PluginUiBase::string_to_float(const string&)’:
../src/plugin_ui_base.cpp:69:80: error: no matching function for call to ‘from_chars(const char*, const char*, float&)’
   69 |   auto [ptr, ec] = std::from_chars(value.data(), value.data() + value.size(), n);
      |                                                                                ^
In file included from ../src/plugin_ui_base.cpp:21:
/usr/include/c++/10.2.0/charconv:595:5: note: candidate: ‘template<class _Tp> std::__detail::__integer_from_chars_result_type<_Tp> std::from_chars(const char*, const char*, _Tp&, int)’
  595 |     from_chars(const char* __first, const char* __last, _Tp& __value,
      |     ^~~~~~~~~~
/usr/include/c++/10.2.0/charconv:595:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/c++/10.2.0/bits/move.h:57,
                 from /usr/include/c++/10.2.0/bits/stl_pair.h:59,
                 from /usr/include/c++/10.2.0/bits/stl_algobase.h:64,
                 from /usr/include/c++/10.2.0/bits/char_traits.h:39,
                 from /usr/include/c++/10.2.0/string:40,
                 from /app/include/glibmm-2.68/glibmm/base64.h:23,
                 from /app/include/glibmm-2.68/glibmm.h:100,
                 from /app/include/giomm-2.68/giomm.h:22,
                 from ../include/plugin_ui_base.hpp:23,
                 from ../src/plugin_ui_base.cpp:20:
/usr/include/c++/10.2.0/type_traits: In substitution of ‘template<bool _Cond, class _Tp> using enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = std::from_chars_result]’:
/usr/include/c++/10.2.0/charconv:584:11:   required by substitution of ‘template<class _Tp> using __integer_from_chars_result_type = std::enable_if_t<std::__or_<std::__or_<std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, signed char>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, short int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, long int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, long long int> >, std::__or_<std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, unsigned char>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, short unsigned int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, unsigned int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, long unsigned int>, std::is_same<typename std::remove_cv< <template-parameter-1-1> >::type, long long unsigned int> >, std::is_same<char, typename std::remove_cv< <template-parameter-1-1> >::type> >::value, std::from_chars_result> [with _Tp = float]’
/usr/include/c++/10.2.0/charconv:595:5:   required by substitution of ‘template<class _Tp> std::__detail::__integer_from_chars_result_type<_Tp> std::from_chars(const char*, const char*, _Tp&, int) [with _Tp = float]’
../src/plugin_ui_base.cpp:69:80:   required from here
/usr/include/c++/10.2.0/type_traits:2554:11: error: no type named ‘type’ in ‘struct std::enable_if<false, std::from_chars_result>’
 2554 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
      |           ^~~~~~~~~~~
[88/105] Compiling C++ object src/easyeffects.p/presets_manager.cpp.o
ninja: build stopped: subcommand failed.
Error: module easyeffects: Child process exited with code 1
```

@Digitalone1 any ideas?

I tried to build the Arch package against this commit, but I'm not familar enough with the PKGBUILD syntax to do so.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,6 @@
   Calf Studio.
 - LV2 filters now are spawned in PipeWire graph only when loaded the first time. Once loaded, they remain connected
   until EasyEffects shutdown.
-- Equalizer APO presets loading is now working properly on locales different than C.
 
 ### Fixed
 
@@ -22,6 +21,7 @@
 - Improved band splitting for crystalizer with new default intensity values.
 - Depending on the input gain or output gain values the corresponding level bars could not be aligned.
 - When adding more equalizer bands they are set to `Bell` instead of `Off`.
+- Equalizer APO presets loading is now working properly on locales different than C.
 - Improved linking management between port filter nodes in PipeWire.
 
 ## [6.0.3]

--- a/include/spectrum_ui.hpp
+++ b/include/spectrum_ui.hpp
@@ -52,7 +52,7 @@ class SpectrumUi : public Gtk::DrawingArea {
 
   std::vector<float> spectrum_mag, spectrum_freqs, spectrum_x_axis;
 
-  std::vector<int> spectrum_bin_count;
+  std::vector<uint> spectrum_bin_count;
 
   void init_color();
 

--- a/src/convolver.cpp
+++ b/src/convolver.cpp
@@ -177,7 +177,7 @@ void Convolver::process(std::span<float>& left_in,
 
     do_convolution(left_out, right_out);
   } else {
-    for (size_t j = 0; j < left_in.size(); j++) {
+    for (size_t j = 0U; j < left_in.size(); j++) {
       data_L.emplace_back(left_in[j]);
       data_R.emplace_back(right_in[j]);
 
@@ -314,9 +314,9 @@ void Convolver::read_kernel_file() {
 
   file.readf(buffer.data(), file.frames());
 
-  for (size_t n = 0; n < buffer_L.size(); n++) {
-    buffer_L[n] = buffer[2 * n];
-    buffer_R[n] = buffer[2 * n + 1];
+  for (size_t n = 0U; n < buffer_L.size(); n++) {
+    buffer_L[n] = buffer[2U * n];
+    buffer_R[n] = buffer[2U * n + 1U];
   }
 
   if (file.samplerate() != static_cast<int>(rate)) {

--- a/src/convolver.cpp
+++ b/src/convolver.cpp
@@ -177,7 +177,7 @@ void Convolver::process(std::span<float>& left_in,
 
     do_convolution(left_out, right_out);
   } else {
-    for (size_t j = 0U; j < left_in.size(); j++) {
+    for (size_t j = 0U, li_size = left_in.size(); j < li_size; j++) {
       data_L.emplace_back(left_in[j]);
       data_R.emplace_back(right_in[j]);
 
@@ -314,7 +314,7 @@ void Convolver::read_kernel_file() {
 
   file.readf(buffer.data(), file.frames());
 
-  for (size_t n = 0U; n < buffer_L.size(); n++) {
+  for (size_t n = 0U, bl_size = buffer_L.size(); n < bl_size; n++) {
     buffer_L[n] = buffer[2U * n];
     buffer_R[n] = buffer[2U * n + 1U];
   }
@@ -379,7 +379,7 @@ void Convolver::set_kernel_stereo_width() {
   float w = static_cast<float>(ir_width) * 0.01F;
   float x = (1.0F - w) / (1.0F + w);  // M-S coeff.; L_out = L + x*R; R_out = R + x*L
 
-  for (uint i = 0U; i < original_kernel_L.size(); i++) {
+  for (uint i = 0U, okl_size = original_kernel_L.size(); i < okl_size; i++) {
     float L = original_kernel_L[i];
     float R = original_kernel_R[i];
 

--- a/src/convolver_ui.cpp
+++ b/src/convolver_ui.cpp
@@ -523,7 +523,7 @@ void ConvolverUi::get_irs_info() {
 
   // rescaling between 0 and 1
 
-  for (size_t n = 0; n < left_mag.size(); n++) {
+  for (size_t n = 0U; n < left_mag.size(); n++) {
     left_mag[n] = (left_mag[n] - min_left) / (max_left - min_left);
     right_mag[n] = (right_mag[n] - min_right) / (max_right - min_right);
   }
@@ -624,21 +624,21 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
   auto log_axis = util::logspace(log10f(20.0F), log10f(22000.0F), spectrum_settings->get_int("n-points"));
   // auto log_axis = util::linspace(20.0F, 22000.0F, spectrum_settings->get_int("n-points"));
 
-  std::vector<int> bin_count(log_axis.size());
+  std::vector<uint> bin_count(log_axis.size());
 
   std::vector<float> l(log_axis.size());
   std::vector<float> r(log_axis.size());
 
   std::ranges::fill(l, 0.0F);
   std::ranges::fill(r, 0.0F);
-  std::ranges::fill(bin_count, 0);
+  std::ranges::fill(bin_count, 0U);
 
   // reducing the amount of data we have to plot and converting the frequency axis to the logarithimic scale
 
-  for (size_t j = 0; j < freq_axis.size(); j++) {
-    for (size_t n = 0; n < log_axis.size(); n++) {
-      if (n > 0) {
-        if (freq_axis[j] <= log_axis[n] && freq_axis[j] > log_axis[n - 1]) {
+  for (size_t j = 0U; j < freq_axis.size(); j++) {
+    for (size_t n = 0U; n < log_axis.size(); n++) {
+      if (n > 0U) {
+        if (freq_axis[j] <= log_axis[n] && freq_axis[j] > log_axis[n - 1U]) {
           l[n] += left_spectrum[j];
           r[n] += right_spectrum[j];
 
@@ -657,10 +657,10 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   // fillint empty bins with their neighbors value
 
-  for (size_t n = 0; n < bin_count.size(); n++) {
-    if (bin_count[n] == 0 && n > 0) {
-      l[n] = l[n - 1];
-      r[n] = r[n - 1];
+  for (size_t n = 0U; n < bin_count.size(); n++) {
+    if (bin_count[n] == 0U && n > 0U) {
+      l[n] = l[n - 1U];
+      r[n] = r[n - 1U];
     }
   }
 

--- a/src/convolver_ui.cpp
+++ b/src/convolver_ui.cpp
@@ -523,7 +523,7 @@ void ConvolverUi::get_irs_info() {
 
   // rescaling between 0 and 1
 
-  for (size_t n = 0U; n < left_mag.size(); n++) {
+  for (size_t n = 0U, lm_size = left_mag.size(); n < lm_size; n++) {
     left_mag[n] = (left_mag[n] - min_left) / (max_left - min_left);
     right_mag[n] = (right_mag[n] - min_right) / (max_right - min_right);
   }
@@ -558,7 +558,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   auto real_input = left_mag;
 
-  for (uint n = 0U; n < real_input.size(); n++) {
+  for (uint n = 0U, ri_size = real_input.size(); n < ri_size; n++) {
     // https://en.wikipedia.org/wiki/Hann_function
 
     auto w = 0.5F * (1.0F - cosf(2.0F * std::numbers::pi_v<float> * n / static_cast<float>(real_input.size() - 1)));
@@ -573,7 +573,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   fftwf_execute(plan);
 
-  for (uint i = 0U; i < left_spectrum.size(); i++) {
+  for (uint i = 0U, ls_size = left_spectrum.size(); i < ls_size; i++) {
     float sqr = complex_output[i][0] * complex_output[i][0] + complex_output[i][1] * complex_output[i][1];
 
     sqr /= static_cast<float>(left_spectrum.size() * left_spectrum.size());
@@ -585,7 +585,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   real_input = right_mag;
 
-  for (uint n = 0U; n < real_input.size(); n++) {
+  for (uint n = 0U, ri_size = real_input.size(); n < ri_size; n++) {
     // https://en.wikipedia.org/wiki/Hann_function
 
     auto w = 0.5F * (1.0F - cosf(2.0F * std::numbers::pi_v<float> * n / static_cast<float>(real_input.size() - 1)));
@@ -595,7 +595,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   fftwf_execute(plan);
 
-  for (uint i = 0U; i < right_spectrum.size(); i++) {
+  for (uint i = 0U, rs_size = right_spectrum.size(); i < rs_size; i++) {
     float sqr = complex_output[i][0] * complex_output[i][0] + complex_output[i][1] * complex_output[i][1];
 
     sqr /= static_cast<float>(right_spectrum.size() * right_spectrum.size());
@@ -615,7 +615,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   freq_axis.resize(left_spectrum.size());
 
-  for (uint n = 0U; n < left_spectrum.size(); n++) {
+  for (uint n = 0U, ls_size = left_spectrum.size(); n < ls_size; n++) {
     freq_axis[n] = 0.5F * static_cast<float>(rate) * static_cast<float>(n) / static_cast<float>(left_spectrum.size());
   }
 
@@ -635,8 +635,8 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   // reducing the amount of data we have to plot and converting the frequency axis to the logarithimic scale
 
-  for (size_t j = 0U; j < freq_axis.size(); j++) {
-    for (size_t n = 0U; n < log_axis.size(); n++) {
+  for (size_t j = 0U, fa_size = freq_axis.size(); j < fa_size; j++) {
+    for (size_t n = 0U, la_size = log_axis.size(); n < la_size; n++) {
       if (n > 0U) {
         if (freq_axis[j] <= log_axis[n] && freq_axis[j] > log_axis[n - 1U]) {
           l[n] += left_spectrum[j];
@@ -657,7 +657,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   // fillint empty bins with their neighbors value
 
-  for (size_t n = 0U; n < bin_count.size(); n++) {
+  for (size_t n = 0U, bc_size = bin_count.size(); n < bc_size; n++) {
     if (bin_count[n] == 0U && n > 0U) {
       l[n] = l[n - 1U];
       r[n] = r[n - 1U];
@@ -678,7 +678,7 @@ void ConvolverUi::get_irs_spectrum(const int& rate) {
 
   // rescaling between 0 and 1
 
-  for (unsigned int n = 0; n < left_spectrum.size(); n++) {
+  for (uint n = 0, ls_size = left_spectrum.size(); n < ls_size; n++) {
     left_spectrum[n] = (left_spectrum[n] - fft_min_left) / (fft_max_left - fft_min_left);
     right_spectrum[n] = (right_spectrum[n] - fft_min_right) / (fft_max_right - fft_min_right);
   }

--- a/src/crossfeed.cpp
+++ b/src/crossfeed.cpp
@@ -83,16 +83,16 @@ void Crossfeed::process(std::span<float>& left_in,
 
   apply_gain(left_in, right_in, input_gain);
 
-  for (size_t n = 0; n < left_in.size(); n++) {
-    data[n * 2] = left_in[n];
-    data[n * 2 + 1] = right_in[n];
+  for (size_t n = 0U; n < left_in.size(); n++) {
+    data[n * 2U] = left_in[n];
+    data[n * 2U + 1U] = right_in[n];
   }
 
   bs2b.cross_feed(data.data(), n_samples);
 
-  for (size_t n = 0; n < left_out.size(); n++) {
-    left_out[n] = data[n * 2];
-    right_out[n] = data[n * 2 + 1];
+  for (size_t n = 0U; n < left_out.size(); n++) {
+    left_out[n] = data[n * 2U];
+    right_out[n] = data[n * 2U + 1U];
   }
 
   apply_gain(left_out, right_out, output_gain);

--- a/src/crossfeed.cpp
+++ b/src/crossfeed.cpp
@@ -83,14 +83,14 @@ void Crossfeed::process(std::span<float>& left_in,
 
   apply_gain(left_in, right_in, input_gain);
 
-  for (size_t n = 0U; n < left_in.size(); n++) {
+  for (size_t n = 0U, li_size = left_in.size(); n < li_size; n++) {
     data[n * 2U] = left_in[n];
     data[n * 2U + 1U] = right_in[n];
   }
 
   bs2b.cross_feed(data.data(), n_samples);
 
-  for (size_t n = 0U; n < left_out.size(); n++) {
+  for (size_t n = 0U, lo_size = left_out.size(); n < lo_size; n++) {
     left_out[n] = data[n * 2U];
     right_out[n] = data[n * 2U + 1U];
   }

--- a/src/crystalizer.cpp
+++ b/src/crystalizer.cpp
@@ -169,7 +169,7 @@ void Crystalizer::process(std::span<float>& left_in,
 
     enhance_peaks(left_out, right_out);
   } else {
-    for (size_t j = 0; j < left_in.size(); j++) {
+    for (size_t j = 0U; j < left_in.size(); j++) {
       data_L.emplace_back(left_in[j]);
       data_R.emplace_back(right_in[j]);
 

--- a/src/crystalizer.cpp
+++ b/src/crystalizer.cpp
@@ -169,7 +169,7 @@ void Crystalizer::process(std::span<float>& left_in,
 
     enhance_peaks(left_out, right_out);
   } else {
-    for (size_t j = 0U; j < left_in.size(); j++) {
+    for (size_t j = 0U, li_size = left_in.size(); j < li_size; j++) {
       data_L.emplace_back(left_in[j]);
       data_R.emplace_back(right_in[j]);
 
@@ -212,7 +212,7 @@ void Crystalizer::process(std::span<float>& left_in,
         notify_latency = true;
       }
 
-      for (uint n = 0U; !deque_out_L.empty() && n < left_out.size(); n++) {
+      for (uint n = 0U, lo_size = left_out.size(); !deque_out_L.empty() && n < lo_size; n++) {
         if (n < offset) {
           left_out[n] = 0.0F;
           right_out[n] = 0.0F;

--- a/src/echo_canceller.cpp
+++ b/src/echo_canceller.cpp
@@ -106,7 +106,7 @@ void EchoCanceller::process(std::span<float>& left_in,
 
   apply_gain(left_in, right_in, input_gain);
 
-  for (size_t j = 0; j < left_in.size(); j++) {
+  for (size_t j = 0U; j < left_in.size(); j++) {
     data_L.emplace_back(left_in[j] * (SHRT_MAX + 1));
     data_R.emplace_back(right_in[j] * (SHRT_MAX + 1));
 

--- a/src/echo_canceller.cpp
+++ b/src/echo_canceller.cpp
@@ -106,7 +106,7 @@ void EchoCanceller::process(std::span<float>& left_in,
 
   apply_gain(left_in, right_in, input_gain);
 
-  for (size_t j = 0U; j < left_in.size(); j++) {
+  for (size_t j = 0U, li_size = left_in.size(); j < li_size; j++) {
     data_L.emplace_back(left_in[j] * (SHRT_MAX + 1));
     data_R.emplace_back(right_in[j] * (SHRT_MAX + 1));
 

--- a/src/effects_base_ui.cpp
+++ b/src/effects_base_ui.cpp
@@ -1103,7 +1103,7 @@ void EffectsBaseUi::setup_listview_selected_plugins() {
           }
         }
       } else {
-        for (size_t m = 0U; m < list.size(); m++) {
+        for (size_t m = 0U, list_size = list.size(); m < list_size; m++) {
           if (list[m] == visible_page_name) {
             listview_selected_plugins->get_model()->select_item(m, true);
 

--- a/src/effects_base_ui.cpp
+++ b/src/effects_base_ui.cpp
@@ -1103,7 +1103,7 @@ void EffectsBaseUi::setup_listview_selected_plugins() {
           }
         }
       } else {
-        for (size_t m = 0; m < list.size(); m++) {
+        for (size_t m = 0U; m < list.size(); m++) {
           if (list[m] == visible_page_name) {
             listview_selected_plugins->get_model()->select_item(m, true);
 

--- a/src/fir_filter_bandpass.cpp
+++ b/src/fir_filter_bandpass.cpp
@@ -40,7 +40,7 @@ void FirFilterBandpass::setup() {
     Creating a bandpass from a band reject through spectral inversion https://www.dspguide.com/ch16/4.htm
   */
 
-  for (size_t n = 0; n < kernel.size(); n++) {
+  for (size_t n = 0U; n < kernel.size(); n++) {
     kernel[n] = lowpass_kernel[n] + highpass_kernel[n];
   }
 

--- a/src/fir_filter_bandpass.cpp
+++ b/src/fir_filter_bandpass.cpp
@@ -40,7 +40,7 @@ void FirFilterBandpass::setup() {
     Creating a bandpass from a band reject through spectral inversion https://www.dspguide.com/ch16/4.htm
   */
 
-  for (size_t n = 0U; n < kernel.size(); n++) {
+  for (size_t n = 0U, k_size = kernel.size(); n < k_size; n++) {
     kernel[n] = lowpass_kernel[n] + highpass_kernel[n];
   }
 

--- a/src/fir_filter_base.cpp
+++ b/src/fir_filter_base.cpp
@@ -98,7 +98,7 @@ auto FirFilterBase::create_lowpass_kernel(const float& cutoff, const float& tran
 
   float sum = 0.0F;
 
-  for (size_t n = 0U; n < output.size(); n++) {
+  for (size_t n = 0U, output_size = output.size(); n < output_size; n++) {
     /*
       windowed-sinc kernel https://www.dspguide.com/ch16/1.htm
     */

--- a/src/fir_filter_base.cpp
+++ b/src/fir_filter_base.cpp
@@ -98,16 +98,16 @@ auto FirFilterBase::create_lowpass_kernel(const float& cutoff, const float& tran
 
   float sum = 0.0F;
 
-  for (size_t n = 0; n < output.size(); n++) {
+  for (size_t n = 0U; n < output.size(); n++) {
     /*
       windowed-sinc kernel https://www.dspguide.com/ch16/1.htm
     */
 
-    if (n == M / 2) {
+    if (n == M / 2U) {
       output[n] = 2.0F * std::numbers::pi_v<float> * fc;
     } else {
-      output[n] = std::sin(2.0F * std::numbers::pi_v<float> * fc * static_cast<float>(n - static_cast<int>(M / 2))) /
-                  static_cast<float>(n - static_cast<int>(M / 2));
+      output[n] = std::sin(2.0F * std::numbers::pi_v<float> * fc * static_cast<float>(n - static_cast<uint>(M / 2U))) /
+                  static_cast<float>(n - static_cast<uint>(M / 2U));
     }
 
     /*

--- a/src/multiband_compressor_ui.cpp
+++ b/src/multiband_compressor_ui.cpp
@@ -545,25 +545,25 @@ void MultibandCompressorUi::reset() {
 }
 
 void MultibandCompressorUi::on_new_frequency_range(std::array<double, n_bands> values) {
-  for (size_t n = 0, m = values.size(); n < m; n++) {
+  for (size_t n = 0U, m = values.size(); n < m; n++) {
     bands_end.at(n)->set_text(level_to_localized_string(values.at(n), 0));
   }
 }
 
 void MultibandCompressorUi::on_new_envelope(std::array<double, n_bands> values) {
-  for (size_t n = 0, m = values.size(); n < m; n++) {
+  for (size_t n = 0U, m = values.size(); n < m; n++) {
     bands_envelope_label.at(n)->set_text(level_to_localized_string(util::linear_to_db(values.at(n)), 0));
   }
 }
 
 void MultibandCompressorUi::on_new_curve(std::array<double, n_bands> values) {
-  for (size_t n = 0, m = values.size(); n < m; n++) {
+  for (size_t n = 0U, m = values.size(); n < m; n++) {
     bands_curve_label.at(n)->set_text(level_to_localized_string(util::linear_to_db(values.at(n)), 0));
   }
 }
 
 void MultibandCompressorUi::on_new_reduction(std::array<double, n_bands> values) {
-  for (size_t n = 0, m = values.size(); n < m; n++) {
+  for (size_t n = 0U, m = values.size(); n < m; n++) {
     bands_gain_label.at(n)->set_text(level_to_localized_string(util::linear_to_db(values.at(n)), 0));
   }
 }

--- a/src/plot.cpp
+++ b/src/plot.cpp
@@ -260,7 +260,7 @@ auto Plot::draw_x_labels(const Cairo::RefPtr<Cairo::Context>& ctx, const int& wi
     would start to be drawn at the border of the window.
   */
 
-  for (size_t n = 0, m = labels.size(); n < m - 1; n++) {
+  for (size_t n = 0U, m = labels.size(); n < m - 1U; n++) {
     Glib::ustring msg;
 
     auto label = labels[n];

--- a/src/plugin_ui_base.cpp
+++ b/src/plugin_ui_base.cpp
@@ -18,7 +18,6 @@
  */
 
 #include "plugin_ui_base.hpp"
-#include <charconv>
 
 PluginUiBase::PluginUiBase(const Glib::RefPtr<Gtk::Builder>& builder,
                            const std::string& schema,
@@ -64,11 +63,15 @@ void PluginUiBase::prepare_spinbutton(Gtk::SpinButton* button, const std::string
 auto PluginUiBase::string_to_float(const std::string& value) -> float {
   // this conversion must be locale independent
 
-  float n;
+  std::stringstream ss;
+  ss.imbue(std::locale::classic());
 
-  auto [ptr, ec] = std::from_chars(value.data(), value.data() + value.size(), n);
+  float fv = 0.0F;
 
-  return (ec == std::errc()) ? n : 0.0F;
+  ss << value;
+  ss >> fv;
+
+  return fv;
 }
 
 void PluginUiBase::set_transient_window(Gtk::Window* transient_window) {

--- a/src/spectrum_ui.cpp
+++ b/src/spectrum_ui.cpp
@@ -106,14 +106,14 @@ void SpectrumUi::on_new_spectrum(const uint& rate, const uint& n_bands, const st
   }
 
   std::ranges::fill(spectrum_mag, 0.0F);
-  std::ranges::fill(spectrum_bin_count, 0);
+  std::ranges::fill(spectrum_bin_count, 0U);
 
   // reducing the amount of data so we can plot them
 
-  for (size_t j = 0, y = spectrum_freqs.size(); j < y; j++) {
-    for (size_t n = 0, m = spectrum_x_axis.size(); n < m; n++) {
-      if (n > 0) {
-        if (spectrum_freqs[j] <= spectrum_x_axis[n] && spectrum_freqs[j] > spectrum_x_axis[n - 1]) {
+  for (size_t j = 0U, y = spectrum_freqs.size(); j < y; j++) {
+    for (size_t n = 0U, m = spectrum_x_axis.size(); n < m; n++) {
+      if (n > 0U) {
+        if (spectrum_freqs[j] <= spectrum_x_axis[n] && spectrum_freqs[j] > spectrum_x_axis[n - 1U]) {
           spectrum_mag[n] += magnitudes[j];
 
           spectrum_bin_count[n]++;
@@ -128,9 +128,9 @@ void SpectrumUi::on_new_spectrum(const uint& rate, const uint& n_bands, const st
     }
   }
 
-  for (size_t n = 0, m = spectrum_bin_count.size(); n < m; n++) {
-    if (spectrum_bin_count[n] == 0 && n > 0) {
-      spectrum_mag[n] = spectrum_mag[n - 1];
+  for (size_t n = 0U, m = spectrum_bin_count.size(); n < m; n++) {
+    if (spectrum_bin_count[n] == 0U && n > 0U) {
+      spectrum_mag[n] = spectrum_mag[n - 1U];
 
       // doing a very poor interpolation for the bins without any count
 

--- a/src/stream_input_effects.cpp
+++ b/src/stream_input_effects.cpp
@@ -166,6 +166,8 @@ void StreamInputEffects::connect_filters() {
 
   auto list = settings->get_string_array("plugins");
 
+  bool mic_linked = false;
+
   uint prev_node_id = pm->input_device.id;
   uint next_node_id = 0U;
 
@@ -182,15 +184,19 @@ void StreamInputEffects::connect_filters() {
 
         auto link_size = links.size();
 
-        for (size_t n = 0; n < link_size; n++) {
+        for (size_t n = 0U; n < link_size; n++) {
           list_proxies.emplace_back(links[n]);
         }
 
-        if (link_size == 2) {
+        if ((mic_linked && (link_size == 2U)) || (!mic_linked && (link_size > 0U))) {
           prev_node_id = next_node_id;
+
+          if (!mic_linked) {
+            mic_linked = true;
+          }
         } else {
           util::warning(log_tag + " link from node " + std::to_string(prev_node_id) + " to node " +
-                        std::to_string(prev_node_id) + " failed");
+                        std::to_string(next_node_id) + " failed");
         }
       }
     }
@@ -212,7 +218,7 @@ void StreamInputEffects::connect_filters() {
     }
   }
 
-  // link spectrum, output level meter and input device
+  // link spectrum, output level meter and source node
 
   auto node_id_list = {spectrum->get_node_id(), output_level->get_node_id(), pm->pe_source_node.id};
 
@@ -223,11 +229,20 @@ void StreamInputEffects::connect_filters() {
 
     auto link_size = links.size();
 
-    for (size_t n = 0; n < link_size; n++) {
+    for (size_t n = 0U; n < link_size; n++) {
       list_proxies.emplace_back(links[n]);
     }
 
-    prev_node_id = (link_size < 2) ? prev_node_id : next_node_id;
+    if ((mic_linked && (link_size == 2U)) || (!mic_linked && (link_size > 0U))) {
+      prev_node_id = next_node_id;
+
+      if (!mic_linked) {
+        mic_linked = true;
+      }
+    } else {
+      util::warning(log_tag + " link from node " + std::to_string(prev_node_id) + " to node " +
+                    std::to_string(next_node_id) + " failed");
+    }
   }
 }
 

--- a/src/stream_output_effects.cpp
+++ b/src/stream_output_effects.cpp
@@ -172,15 +172,15 @@ void StreamOutputEffects::connect_filters() {
 
         auto link_size = links.size();
 
-        for (size_t n = 0; n < link_size; n++) {
+        for (size_t n = 0U; n < link_size; n++) {
           list_proxies.emplace_back(links[n]);
         }
 
-        if (link_size == 2) {
+        if (link_size == 2U) {
           prev_node_id = next_node_id;
         } else {
           util::warning(log_tag + " link from node " + std::to_string(prev_node_id) + " to node " +
-                        std::to_string(prev_node_id) + " failed");
+                        std::to_string(next_node_id) + " failed");
         }
       }
     }
@@ -202,9 +202,9 @@ void StreamOutputEffects::connect_filters() {
     }
   }
 
-  // link spectrum, output level meter and output device
+  // link spectrum and output level meter
 
-  auto node_id_list = {spectrum->get_node_id(), output_level->get_node_id(), pm->output_device.id};
+  auto node_id_list = {spectrum->get_node_id(), output_level->get_node_id()};
 
   for (const auto& node_id : node_id_list) {
     next_node_id = node_id;
@@ -213,11 +213,33 @@ void StreamOutputEffects::connect_filters() {
 
     auto link_size = links.size();
 
-    for (size_t n = 0; n < link_size; n++) {
+    for (size_t n = 0U; n < link_size; n++) {
       list_proxies.emplace_back(links[n]);
     }
 
-    prev_node_id = (link_size < 2) ? prev_node_id : next_node_id;
+    if (link_size == 2U) {
+      prev_node_id = next_node_id;
+    } else {
+      util::warning(log_tag + " link from node " + std::to_string(prev_node_id) + " to node " +
+                    std::to_string(next_node_id) + " failed");
+    }
+  }
+
+  // link output device
+
+  next_node_id = pm->output_device.id;
+
+  auto links = pm->link_nodes(prev_node_id, next_node_id);
+
+  auto link_size = links.size();
+
+  for (size_t n = 0U; n < link_size; n++) {
+    list_proxies.emplace_back(links[n]);
+  }
+
+  if (link_size < 2U) {
+    util::warning(log_tag + " link from node " + std::to_string(prev_node_id) + " to output device " +
+                  std::to_string(next_node_id) + " failed");
   }
 }
 

